mozilla/bedrock
# Automate Root Store Policy page generation and fix missing indentation of nested lists

## Summary

This changeset adds a tool to automatically refresh the Root Store Policy page, which is drawn from source in markdown in a seperate repo.

Up until now, updates to this were done manually, and with a Markdown parser that (as reported in #14374 ) had some conversion issues that meant some data was missing.

This changeset switches to a new parser (commonmark) as part of the work.

## Significant changes and points to review

* The diff of the source has a number of changes in it, but comparing the rendered output text (see [this gist](https://gist.github.com/stevejalim/780d40efc20a76a04562e94c59ce0055)) shows no significant change other than the addition of an extra TOC link (automatically pulled from the document as it's a `h2` and possibly overlooked in the manual process)
* hyperlinks are absolute, not relative and not generated with the jinja2 `url` tag. However, I think this is acceptable as this doc only exists in en-US English and so it's not worth further yak-shaving to turn it into a template tag.

Also, as raised in #14374, we now properly next sublists:

**Before**

![](https://private-user-images.githubusercontent.com/24921711/318241985-8f41a96c-d4d2-4cad-8752-a9d41ff451b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNDU5ODUsIm5iZiI6MTcxMjE0NTY4NSwicGF0aCI6Ii8yNDkyMTcxMS8zMTgyNDE5ODUtOGY0MWE5NmMtZDRkMi00Y2FkLTg3NTItYTlkNDFmZjQ1MWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDEyMDEyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwNGYzNDgyODI3ZjNkNTdjYzg3OGM5YTkzNWQzZmFhOWJmNzAwYTlhYTAyZTc3Nzc0YTBjZTIyZDE0ZDg4NWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SmdDg-EzjUjMfnmC8qZA7G2ea2mUlthhHQ107Np_U5c)

**After**
![](https://private-user-images.githubusercontent.com/101457/319174179-5689af99-92f8-49cb-b452-1f99799fc8e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNDYxNTMsIm5iZiI6MTcxMjE0NTg1MywicGF0aCI6Ii8xMDE0NTcvMzE5MTc0MTc5LTU2ODlhZjk5LTkyZjgtNDljYi1iNDUyLTFmOTk3OTlmYzhlOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwM1QxMjA0MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mOTM4YjhiMzViYzE3ZWM0NmFiYjMyNDEyZmRmOWNkYjdiNjg2ZTYwY2QyYjZkYjdmMGYyZTgwNDg3ZTIyZThmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KzFU54so4CxzHGFmv8PLwCPBrXu87Q6GCfV5ZXhcfp4)

## Issue / Bugzilla link

Resolves #14374 

## Testing

* Pull the branch locally
* `make preflght`
* Delete or deface the HTML in `bedrock/mozorg/templates/mozorg/about/governance/policies/security/certs/policy.html`
* run `./manage.py update_root_store_policy_page` (with the `--dry-run` flag first if you like
* After running the command in standard/not-dry-run mode, confirm that the policy doc is reinstated (i.e., does not show changes with `git status`)
* `npm run start` to run locally
* Compare http://localhost:8000/en-US/about/governance/policies/security-group/certs/policy/ with https://www.mozilla.org/en-US/about/governance/policies/security-group/certs/policy/ including markup of hyperlinks
